homeassistant/components/netatmo/__init__.py

```diff
@@ -79,6 +88,30 @@
             hass.config_entries.async_forward_entry_setup(entry, component)
         )
 
+    if CONF_WEBHOOK_ID not in entry.data:
+        entry.data[CONF_WEBHOOK_ID] = secrets.token_hex()
+        hass.config_entries.async_update_entry(entry, data=entry.data)
+
+    if hass.components.cloud.async_active_subscription():
+        webhook_url = await hass.components.cloud.async_create_cloudhook(
```

Comment:
Cloudhooks are persisted, so you should only do it once and also store it in entry.data.